src/Wig.Zstd/Program.cs

```diff
@@ -29,36 +35,143 @@ public sealed class DefaultCommand : AsyncCommand<DefaultCommand.Settings>
 
-            [CommandOption("-s|--source")]
-            [Description("Show example source code")]
-            public bool Source { get; set; }
+            [CommandOption("-o|--overwrite")]
+            [Description("Overwrite file / directory")]
+            public bool Overwrite { get; set; }
 
-            [CommandOption("--select")]
-            [Description("Show example source code")]
-            public bool Select { get; set; }
+            [CommandOption("-l|--level")]
+            [Description("Compression level (1-22)")]
+            public int CompressionLevel { get; set; }
+
+            [CommandOption("-s|--subfolder")]
+            [Description("Subfolder name")]
+            public string Subfolder{ get; set; }
         }
 
         public DefaultCommand(IAnsiConsole console)
         {
             _console = console;
         }
 
+        public async Task WriteFileAsync(byte[] data, string path, string subfolder, string extension)
+        {
+            var fileName = Path.GetFileName(path);
+            var currentDirectory = Path.GetDirectoryName(path);
+
+            var directory = Directory.CreateDirectory($"{currentDirectory}/{subfolder}");
+
+            await using (FileStream fstream = new FileStream($"{directory}/{fileName}{extension}", FileMode.OpenOrCreate))
+            {
+                await fstream.WriteAsync(data, 0, data.Length);
+            }
+        }
+
+        public async Task CompressingAsync(string path, int compressionLevel, bool overwrite, string subfolder)
+        {
+            using var options = new CompressionOptions(compressionLevel);
+            using var compressor = new Compressor(options);
+            FileAttributes attr = File.GetAttributes(path);
+            if (attr.HasFlag(FileAttributes.Directory))
+            {
+                string[] filePaths = Directory.GetFiles(path);
+
+                foreach (var filePath in filePaths)
+                {
+                    byte[] data = await File.ReadAllBytesAsync(filePath);
+
+                    var compData = compressor.Wrap(data);
+
+                    if (File.Exists($"{filePath}.zs") && !overwrite)
+                    {
+                        AnsiConsole.WriteLine("A compressed file with the same name already exists. Use the -o | --overwrite parameter to force overwrite.");
+                        return;
+                    }
+
+                    await WriteFileAsync(compData, path, subfolder, "zs");
+                }
+            }
+
+            byte[] sourceData = await File.ReadAllBytesAsync(path);
+
+            var compressedData = compressor.Wrap(sourceData);
+
+            if (File.Exists($"{path}.zs") && !overwrite)
+            {
+                AnsiConsole.WriteLine("A compressed file with the same name already exists. Use the -o | --overwrite parameter to force overwrite.");
+                return;
+            }
+
+            await WriteFileAsync(compressedData, path, subfolder, ".zs");
```

Comment:
Here extension parameter is ".zs" but on the line 102 its a "zs". What's difference?
We also using string.Empty for decompression, should we just have overload without extension parameter for WriteFileAsync or default value for it (string.Empty) instead of passing string.Empty on call?